packages/nimble-components/src/table/components/row/tests/table-row.spec.ts

```diff
@@ -221,6 +222,48 @@ describe('TableRow', () => {
 
             expect(listener.spy).not.toHaveBeenCalled();
         });
+
+        it('shows expand-collapse button when isParentRow is true', async () => {
+            const pageObject = new TableRowPageObject(element);
+            await connect();
+            element.isParentRow = true;
+            await waitForUpdatesAsync();
+
+            expect(pageObject.getExpandCollapseButton()).toBeDefined();
+        });
+
+        it('hides expand-collapse button when isParentRow is false', async () => {
+            const pageObject = new TableRowPageObject(element);
+            await connect();
+            element.isParentRow = true;
+            await waitForUpdatesAsync();
+
+            element.isParentRow = false;
+            await waitForUpdatesAsync();
+            expect(pageObject.getExpandCollapseButton()).toBeNull();
+        });
+
+        it('toggling expand-collapse button fires "row-expand-toggle" event', async () => {
+            const pageObject = new TableRowPageObject(element);
+            await connect();
+            element.isParentRow = true;
+            element.recordId = 'foo';
+            await waitForUpdatesAsync();
+            const expandCollapseButton = pageObject.getExpandCollapseButton();
+
+            const listener = createEventListener(element, 'row-expand-toggle');
+            expandCollapseButton!.click();
+            await listener.promise;
+
+            expect(listener.spy).toHaveBeenCalledTimes(1);
+            const expectedDetails: TableRowExpandToggleEventDetail = {
+                newState: true,
```

Comment:
Can you assert the value of `element.expanded` before and after clicking the button too?

Reply:
Unfortunately, no, at least as a unit test isolated to the `Row` component. This is admittedly a goofy concession with how the API ultimately works (which is also how the `GroupRow` works as well). The `expanded` API is controlled externally to the `Row` instance based on Tanstack-provided state. When we click the button, the toggle event is ultimately handled by the `Table` which then updates Tanstack, which, in turn, provides a new set of row state that is then set on the `Row` instances, which would include an updated value for the `expanded` state.